enhancements/authentication/invalid-certs.md

```diff
@@ -0,0 +1,152 @@
+---
+title: invalid-certs
+authors:
+  - "@s-urbaniak"
+reviewers:
+  - "@sttts"
+  - "@stlaz"
+  - "@deads2k"
+approvers:
+  - "@mfojtik"
+creation-date: 2022-12-08
+last-updated: 2022-12-08
+---
+
+# Detecting invalid CA certificates
+
+## Summary
+
+OpenShift 4.10 is going to be rebased against Kubernetes 1.23. This requires using Go 1.17.
+However, starting with Go 1.17 support for legacy invalid is going to be removed, see https://go.dev/doc/go1.17.
+More formally, the temporary `GODEBUG=x509ignoreCN=0` flag has been removed.
+This implies that starting from OpenShift 4.10 invalid certificates will not be trusted any more
+as they will fail verification.
+
+This enhancement suggests a solution to prevent broken workloads from OpenShift 4.9 to OpenShift 4.10
+if invalid certificates are detected.
+
+This enhancment therefore proposes:
+- a way to detect invalid certificates in kube-apiserver and oauth-server
+- a way to prevent upgrades from OpenShift 4.9 to OpenShift 4.10 in the face of invalid certificates
+
+## Motivation
+
+We want to prevent broken workloads due to invalid certificates
+when upgrading from OpenShift 4.9 to OpenShift 4.10.
+
+### Goals
+
+A means to detect invalid certificates for TLS connections between:
+- kube-apiserver and API webhooks
+- kube-apiserver and aggregated API endpoints
+- oauth-server and its external route
+- oauth-server and external identity providers
+
+### Non-Goals
+
+Other core workloads are out of scope for this enhancement.
```

Comment:
do we at least know what the other workloads are that are likely impacted by this?

does every component that exposes a TLS metrics endpoint need to do work, for example?

Reply:
I am not aware that all components expose a TLS metric like this. Our finding refers concretely to a contribution that was made upstream to Kubernetes.

An initial search on the openshift org reveals quite a few usages: https://github.com/search?q=org%3Aopenshift+GODEBUG%3Dx509ignoreCN%3D0&type=code

maybe @eparis has more insight as I believe he had touch points with the `GODEBUG=x509ignoreCN=0` setting.